instrumentation/pulsar/pulsar-2.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/telemetry/PulsarSingletons.java

```diff
@@ -136,7 +137,8 @@ private static Instrumenter<PulsarRequest, Void> createProducerInstrumenter() {
             .addAttributesExtractor(
                 createMessagingAttributesExtractor(getter, MessageOperation.PUBLISH))
             .addAttributesExtractor(
-                ServerAttributesExtractor.create(new PulsarNetClientAttributesGetter()));
+                ServerAttributesExtractor.create(new PulsarNetClientAttributesGetter()))
+            .addOperationMetrics(MessagingProducerMetrics.get());
```

Comment:
This could apply to other messaging instrumentations too, just adding it to pulsar is fine if you are not interested in working on them or wish to keep the PR small so it would be easier to review.
You should modify pulsar tests to verify the metric there also.

Reply:
Yes, it can be easier to review. 
Some of the other messaging systems are not very familiar to me. I will try to improve some other messaging systems in the future.

Reply:
> You should modify pulsar tests to verify the metric there also.

Done; please take a look. Thanks.